genpolicy: support ephemeral volume source

burgerdev opened the pull request:
The Ephemeral volume type is more or less a managed version of the PersitentVolumeClaim volume type. Therefore, it can be supported with the rules for PVCs, and only needs deserialization support.

---

The use case is close enough that I decided to just add fields to the existing PVC policy test, instead of creating a new test file.

cc @Redent0r @danmihai1 

---

Btw, I noticed that `get_shared_bind_mount` has `access` and `propagation` both in the `yaml_mount` as well as in explicit arguments, but only the latter are used (and those are hard-coded). Is that WAI?

burgerdev commented:
Sorry, I had already forgotten about that. My use case is ephemeral *devices*, for which we already have the policy implemented, so I really only need the additional struct field. Would you mind carving that out from your PR?

sprt commented:
Feel free to cherry-pick from my PR and adapt it - that should not block your work.

burgerdev commented:
So, I could cherry-pick https://github.com/kata-containers/kata-containers/pull/10559/commits/a32371d8433a59e69ea00b1a9477f04cd02beaee and split it into the parts that allow plain ephemeral volumes and the parts that add policy and fixed options per storage class. Then, change this PR here to the former part. wdyt?

sprt commented:
Yep SGTM!
